wp-admin/js/widgets/media-gallery-widget.js

```diff
@@ -88,11 +88,44 @@
 		 * @returns {void}
 		 */
 		renderPreview: function renderPreview() {
-			var control = this, previewContainer, previewTemplate;
+			var control = this, previewContainer, previewTemplate, attachments;
+
 			previewContainer = control.$el.find( '.media-widget-preview' );
 			previewTemplate = wp.template( 'wp-media-widget-gallery-preview' );
-			previewContainer.html( previewTemplate( control.previewTemplateProps.toJSON() ) );
+
+			if ( control.model.get( 'ids').length && ! control.model.get( 'attachemnts' ) ) {
+				attachments = this.getAttachments();
+
+				attachments.more().done( function() {
+					control.model.set( 'attachments', JSON.stringify( _.pluck( attachments.models, 'attributes' ) ) );
+					previewContainer.html( previewTemplate( control.model.attributes ) );
+				} );
+			} else {
+				previewContainer.html( previewTemplate( control.previewTemplateProps.toJSON() ) );
+			}
 		},
+
+		/**
+		 * Fetch attachment models.
+		 *
+		 * @returns {wp.media.model.Attachments} A Backbone.Collection.
+		 */
+		getAttachments: function getAttachments() {
+			var attachments,
+				ids = this.model.get( 'ids' ).split( ',' );
```

Comment:
The underlying widget instance can store an array instead of a string.

Reply:
👍 Will play with that idea.

Reply:
Actually, if we were to do that, then we'll have to convert the array to a string on the front end before rendering the shortcode. Not sure which is preferred?

Reply:
@joemcgill you mean when `ids` is passed into `gallery_shortcode()`?  I can see that this will get mapped to the `include` arg which is passed to `get_children()` and then down into `get_posts()` which will then pass it into `wp_parse_id_list( $r['include'] )`, which accepts both an array and a string. So I don't think it will get converted to a string?